llvm/lib/Target/AMDGPU/GCNHazardRecognizer.cpp

```diff
@@ -935,6 +935,30 @@ int GCNHazardRecognizer::checkVALUHazards(MachineInstr *VALU) {
           if (Use.isReg() && TRI->regsOverlap(Def, Use.getReg()))
```

Comment:
Why not just include implicit operands in this loop? I.e. iterate over `all_uses` instead of `explicit_uses`.

Reply:
In the past, I have seen us use implicit regs on ops where they don't actually read those regs (rather they're use for modelling liveness in some way, MI100 indirect copy to AGPR handling comes to mind). I assume we don't do this / plan on doing this for MI300 (at least for VALUs)?

Reply:
> I assume we don't do this / plan on doing this for MI300 (at least for VALUs)?

I can't promise that, but at least scanning all implicit operands would be correct by default. If necessary you could ignore certain implicit operands if you are sure it is still safe. What you are doing in this patch is the opposite of that.

Reply:
Most notably this will break on inline asm uses

Reply:
We check uses for users that SIInstrInfo::isVALU